src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -580,10 +583,9 @@ def fill_ldap_cache(self, force=False):
 
     @private
     async def get_ldap_cache(self):
-        """
-        """
         if not await self.middleware.call('cache.has_key', 'LDAP_cache'):
-            await self.middleware.run_in_thread(self.fill_ldap_cache)
+            cache_job = await self.middleware.run_in_thread(self.fill_ldap_cache)
```

Comment:
I am not sure this works as expected, proper way would be `await self.middleware.call('ldap.fill_ldap_cache')`